tests/python_tests/test_preemption.py

```diff
@@ -69,36 +94,73 @@ def test_preemption_with_multinomial(tmp_path, dynamic_split_fuse):
+                " status of all the other passengers?\nWe're the only ones left, so no...\nI don't think they'll really leave.\nThey"
+            ],
+            [
+                "\nI don't have any knowledge on them. We are based out near Dubai so hopefully they will take care of us soon enough :) thanks though :",
+                "\nUAE is not one of the richest countries in Asia but definitely among those most corrupt nations because this corruption (and its own endemic practices) still",
+                "\nNope, I'm just going through my first semester there right now and it was nice to see some people who were doing well haha - we",
+                "\nIt's a country where your parents can never give you anything at all!  It also has an extremely low education system for many years... You",
+            ],
         ],
-        [
-            "\nIt's in the middle of nowhere if you haven’t seen one yet! It might be more convenient there than anywhere else.. maybe take",
-            '\nUAE is a country with some great culture that has been living under Islamic oppression for almost 60 years now (including 20 years as part of Arab',
-            '\nNope, just wanted to say how awesome and beautiful it was when my brother came back from an adventure trip across Asia - our 2nd year',
-            '\nI don\'t know anything.  I\'m not sure what kind this sub wants though... but apparently they are pretty bad at making videos/photos'
+        darwin=[
+            [
+                "\nI've had a friend with the capacity to test this in his own words.\nThe big problem with real-world results is the rigidity"
+            ],
+            [
+               ' position of the patent application number of the present invention?\n\nIn the present invention, the present invention relates to an improved method for manufacturing a semic',
+               ' status of your town? How many houses do you have?\nThere are about three houses in our town. The closest place to us is about 25',
+               " status of all the other passengers?\nWe're the only ones left, so no...\nI don't think they'll really leave.\nThey"
+            ],
+            [
+                "\nI don't have any knowledge on them. We are based out near Dubai so hopefully they will take care of us soon enough :) thanks though :",
+                "\nUAE is not one of the richest countries in Asia but definitely among those most corrupt nations because this corruption (and its own endemic practices) still",
+                "\nNope, I'm just going through my first semester there right now and it was nice to see some people who were doing well haha - we",
+                "\nIt's a country where your parents can never give you anything at all!  It also has an extremely low education system for many years... You",
+            ],
         ],
-    ])
+    ).get_ref_texts(),
```

Comment:
You always call `get_ref_texts()` immediately after creating `RandomSamplingRefTexts`. Its instance isn't saved. That means it should be just a single function call. But in that particular case the simpler solution is to use dict:

```py
ref_texts={
        "linux": [...],
        "win32": [ ...],
        "darwin": [ ...],
    }[sys.platform]
```

Apply it to tests/python_tests/test_sampling.py as well.

Reply:
Is there a reason not to use dictionary?

Reply:
Never mind